src/qililab/qprogram/decorators.py

```diff
@@ -18,9 +18,7 @@ def wrapper(*args, **kwargs):
             if parameter not in kwargs and len(args) == 1:
                 kwargs[parameter] = None
             # Get the argument by name
-            param_value = (
-                kwargs.get(parameter) if parameter in kwargs else args[1]
-            )  # assuming 1 is the default position
+            param_value = kwargs.get(parameter) if parameter in kwargs else None
```

Comment:
why are we changing this from args[1] to None?

Reply:
Because before we always tried to get the first parameter (args[1]) which is wrong. Now, if `parameter` is not there, the `isinstance(None, Variable)` will not raise the `ValueError` falsely.